Implement databag fallback (#32)
inspec/inspec-vault: README.md

In this case, Chef InSpec searches the `secret/configuration/webserver` document and returns the value of the `password` key.

> ## Usage with TestKitchen

Reviewer: ```suggestion
## Usage with Test Kitchen
```

> To allow for more dev/prod parity, this input plugin detects if it is called from within TestKitchen. As tests should limit access to third party systems, by default the plugin will revert on using the `data_bags_path` from kitchen's `provisioner` section:

Reviewer: ```suggestion
To allow for more dev/prod parity, this input plugin detects if it is called from within Test Kitchen. As tests should limit access to third party systems, by default the plugin will revert on using the `data_bags_path` from kitchen's `provisioner` section:
```

Reviewer 2: That one always slips through, thanks for spotting it

> To allow for more dev/prod parity, this input plugin detects if it is called from within Test Kitchen. As tests should limit access to third party systems, by default the plugin will revert on using the `data_bags_path` from kitchen's `provisioner` section:

Reviewer 3: ```suggestion
To allow for more development/production parity, this input plugin detects if it is called from within Test Kitchen. As tests should limit access to third party systems, by default the plugin will revert on using the `data_bags_path` from kitchen's `provisioner` section:
```

```yaml
suites:
  - name: default
    verifier:
      load_plugins: true
    data_bags_path: "test/integration/data_bags"
```

With this configuration, the databag at `test/integration/data_bags/inspec` will be accessed and the contents of the `vault.json` file within this directory get parsed. Any Vault lookups will be evaluated against the contained data.

The databag fallback mode and the databag or item names can be configured for the plugin.

> Please note, that support for `load_plugins` was introduced in version 1.3.2 of the `kitchen-inspec` verifier plugin. Earlier versions are unable to load InSpec V2 plugins.

Reviewer 3: ```suggestion
Support for `load_plugins` was introduced in version 1.3.2 of the `kitchen-inspec` verifier plugin. Earlier versions are unable to load InSpec V2 plugins.
```

This plugin supports the following options:

### INSPEC_DATABAG_FALLBACK

### databag_fallback

> A boolean that indicates if the plugin should use a data bag within Test Kitchen; default value is "true". This allows mocking a Vault server in development instances.

Reviewer 3: ```suggestion
A boolean that indicates if the plugin should use a data bag within Test Kitchen. The default value is "true." This allows for mocking a Vault server in development instances.
```

### INSPEC_DATABAG_ITEM

### databag_item

> A string with the name of the data bag item to use, if `databag_fallback` is true; default value is "vault".

Reviewer 3: ```suggestion
A string with the name of the data bag item to use. If `databag_fallback` is `true`, then the default value is "vault."
```

### INSPEC_DATABAG_NAME

### databag_name

> A string with the name of the data bag to use, if `databag_fallback` is true; default value is "inspec".

Reviewer 3: ```suggestion
A string with the name of the data bag to use. If `databag_fallback` is `true`, then the default value is "inspec".
```
